matrix/src/dense.rs

```diff
@@ -383,3 +384,147 @@ impl<T: Clone> MatrixRowSlicesMut<T> for RowMajorMatrixViewMut<'_, T> {
         &mut self.values[r * self.width..(r + 1) * self.width]
     }
 }
+
+impl<T> MatrixTranspose<T> for RowMajorMatrix<T>
+where
+    T: Clone + Default + Send + Sync,
+{
+    const BLOCK_SIZE: usize = match core::mem::size_of::<T>() {
+        4 => 64,
+        8 => 32,
+        16 => 16,
+        _ => 16,
+    };
+
+    fn transpose(self) -> Self {
+        let block_size = Self::BLOCK_SIZE;
+        let height = self.height();
+        let width = self.width();
+
+        let transposed_values: Vec<T> = (0..width * height).map(|_| T::default()).collect();
```

Comment:
nit: could this be `vec![T::default(); width * height]`?

Reply:
Yup, much better !